### PR TITLE
feat(FX-3112): edit saved search alert

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves [FX-3112]

### Description
* As an user with the saved search M2 feature flag enabled,
* I can navigate to the profile screen
* And see a link "saved alerts"
* And tapping that link takes me to a screen where all of my saved alerts are listed
* And tapping one of the listed alerts takes me to a form where I can update the name of my saved search
* And user go back and sees the updated name in the list of saved search alerts

### Acceptance criteria:
* Screen title is the name of the saved alert
* “Save alert” triggers updateSavedSearch mutation and passes name param, then navigates back to the list of saved searches
* Update the old name in the list of saved searches
* Not included: “remove” button and functionality

### Useful information
When the user navigate go back to the list of saved search alert. The list [scrolls up](https://github.com/artsy/eigen/blob/c9f1064d4849f341ef38cdec0142f1d0be617421/src/lib/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tsx#L27) the page since the refetch request is executed. This is a temporary solution, in the future it will be possible to update the name without making a refetch. More context can be [found here](https://artsy.slack.com/archives/CP9P4KR35/p1628183548076500?thread_ts=1628180088.071300&cid=CP9P4KR35)

#### iOS
https://user-images.githubusercontent.com/3513494/128513550-3d40471f-65bf-43fe-99f6-6383c51580e2.mp4

### Android
https://user-images.githubusercontent.com/3513494/128532347-da6c127a-2358-4f3a-b91f-7dbe9a3037be.mp4

TODO: clarify form error states and criteria

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Edit saved search alert screen - dzmitry tratsiak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[FX-3112]: https://artsyproduct.atlassian.net/browse/FX-3112

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8f32b469423048432839528525588bbfeb880ccf
</p>

**Review by iskounen — APPROVED** [review]
I have one little question, but these changes look good to me 👍

File: src/lib/Scenes/MyProfile/MyProfile.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ const MyProfile: React.FC<{ me: MyProfile_me; relay: RelayRefetchProp }> = ({ me
         <MenuItem isBeta={true} title="My Collection" onPress={() => navigate("my-collection")} />
       )}
       <MenuItem title="Saves and follows" onPress={() => navigate("favorites")} />
-      {!!showSavedSearchV2 && <MenuItem title="Saved Alerts" onPress={() => navigate("saved-search-alerts/list")} />}
+      {!!showSavedSearchV2 && <MenuItem title="Saved Alerts" onPress={() => navigate("/my-profile/saved-search-alerts")} />}
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/lib/Scenes/MyProfile/MyProfile.tsx]: Yes, you're right.

**Review by dblandin — APPROVED** [review]
Looks great!! 💯

**Pull request merged**